classes/class-wc-rest-connect-shipping-rates-controller.php

```diff
@@ -38,9 +38,11 @@ private function has_customs_data( $package ) {
 	 * @return array|WP_Error
 	 */
 	public function post( $request ) {
-		$payload = $request->get_json_params();
+		$payload                        = $request->get_json_params();
 		$payload[ 'payment_method_id' ] = $this->settings_store->get_selected_payment_method_id();
-		$order_id = $request[ 'order_id' ];
+		$order_id                       = $request[ 'order_id' ];
+		// Don't add extra rates if request is coming from Calypso ( Store on WP.com ).
+		$add_extra_rates_to_request     = ! isset( $request['_via_calypso'] );
```

Comment:
Can the request coming in specify what version it wants? something like `$request['wcs_version']` or something? 
Then we can check for version like
`if (version_compare(CURRENT_WCS_VERSION, `1.22.0`, '>=') === true)`

Reason why I am asking is cause I think `_via_calypso` will always stay `true` from them, so now the condition becomes "if it is from wp.com calypso store, use 1.21 response"; versus "if request coming in is 1.21, we response in 1.21". 

[updated]
Just checked with @c-shultz , there is no `wcs_version` we can use :(

Reply:
That said, I think this will fix the existing problem though.